source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -506,6 +507,9 @@ bool CompressorFilterConfig::DirectionConfig::isMinimumContentLength(
     }
     return is_minimum_content_length;
   }
+  if (always_compress_content_if_exists_.enabled()) {
+    return true;
+  }
```

Comment:
This runtime flag is enabled by default which changes the result of this function for existing installations working with HTTP/1.1. Not sure this is a good idea.

If we aim to fix H2 streaming then a better approach would be to narrow the `Transfer-Encoding` check below to apply to HTTP/1.1 only IMHO. `Transfer-Encoding` was introduced in HTTP/1.1 and its `chunked` value was dropped in HTTP2. WDYT?

Reply:
i will rework so that default behavior wont be changed and if set to false, it would ignore transfer-encoding chunked and return true unconditionally. how thats sounds @rojkov ?